ui/design-system-core/src/components/Box/index.tsx

```diff
@@ -1,14 +1,24 @@
 import React, { PropsWithChildren } from 'react';
-import { tw } from '@twind/core';
+import { apply, tw } from '@twind/core';
 
 export interface IBoxProps {
   customStyle?: string;
+  onClick?: () => void;
```

Comment:
```suggestion
```
Why `onClick` here ? You can use the semantically relevant button component in this case with plain prop as `<Button onClick={...} plain> ... </Button>`.  You can [read](https://www.builder.io/blog/buttons) here why we prefer handling button over div for handling `onClick` events.

Reply:
Alright. Will take a look